### ISSUE TITLE
DeprecationWarning: BaseReporter using __implements__

### ISSUE THREAD

**Issue by rgilton** [carsongee/pytest-pylint#167]
When I run `pytest --pylint`, I get this warning:

```
../../../../../../.local/share/virtualenvs/server-IJExtVut/lib/python3.10/site-packages/pylint/reporters/base_reporter.py:35
  /home/rob/.local/share/virtualenvs/server-IJExtVut/lib/python3.10/site-packages/pylint/reporters/base_reporter.py:35: DeprecationWarning: Using the __implements__ inheritance pattern for BaseReporter is no longer supported. Child classes should only inherit BaseReporter
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
```
